### PR TITLE
Move all libraries into lib

### PR DESCRIPTION
**Pull request by wmmiii**
Everything except for //software/devices and GTest code has been
moved under //lib.

### CODE REVIEW

**Comment by wmmiii** [comment]
@ademuri I posted this so we can start to discuss the structure. Note that the node build is not working with the following class of compiler errors:

```
In file included from .pio/libdeps/node/RadioHead/RHCRC.h:12:0,
                 from .pio/libdeps/node/RadioHead/RHCRC.cpp:43:
.pio/libdeps/node/RadioHead/RadioHead.h:1501:12: fatal error: SPI.h: No such file or directory

*************************************************************
* Looking for SPI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:SPI.h"
* Web  > https://registry.platformio.org/search?q=header:SPI.h
*
*************************************************************
```

I feel like we've run into this before. Do you recall the solution?

**Comment by wmmiii** [comment]
I got past the SPI issue by explicitly requesting it in platformio.ini. However now it's angry about FastLED, but it's late, and I'm off to bed.

**Comment by wmmiii** [comment]
@ademuri  Also, when should I be using `#include <angle_includes>` and when not? I'm pretty fuzzy on their use in this project.

File: lib/arduino/FastLedManager.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include "FastLedManager.hpp"
 
-#include "../generic/DeviceDescription.hpp"
+#include "../device/DeviceDescription.hpp"
```
Comment (ademuri) [inline lib/arduino/FastLedManager.cpp]: You should use the `#include <DeviceDescription.hpp>` style for anything which is in `lib/`.

**Review by ademuri — COMMENTED** [review]
This works in [colordance-circle](https://github.com/ademuri/colordance-circle/blob/master/lib/generic/ColordanceTypes.hpp), I would look there to see what's different.

**Comment by wmmiii** [comment]
Closing this in lieu of #30 and its follow-ups

**Pull request closed without merging**